checks/evaluation/cii_best_practices_test.go

```diff
@@ -16,71 +16,119 @@ package evaluation
 import (
 	"testing"
 
-	"github.com/ossf/scorecard/v4/checker"
-	"github.com/ossf/scorecard/v4/clients"
+	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/finding"
+	scut "github.com/ossf/scorecard/v4/utests"
 )
 
 func TestCIIBestPractices(t *testing.T) {
-	t.Run("CIIBestPractices", func(t *testing.T) {
-		t.Run("in progress", func(t *testing.T) {
-			r := &checker.CIIBestPracticesData{
-				Badge: clients.InProgress,
-			}
-			result := CIIBestPractices("CIIBestPractices", nil, r)
-			if result.Score != inProgressScore {
-				t.Errorf("CIIBestPractices() = %v, want %v", result.Score, inProgressScore)
-			}
-		})
-		t.Run("passing", func(t *testing.T) {
-			r := &checker.CIIBestPracticesData{
-				Badge: clients.Passing,
-			}
-			result := CIIBestPractices("CIIBestPractices", nil, r)
-			if result.Score != passingScore {
-				t.Errorf("CIIBestPractices() = %v, want %v", result.Score, passingScore)
-			}
-		})
-		t.Run("silver", func(t *testing.T) {
-			r := &checker.CIIBestPracticesData{
-				Badge: clients.Silver,
-			}
-			result := CIIBestPractices("CIIBestPractices", nil, r)
-			if result.Score != silverScore {
-				t.Errorf("CIIBestPractices() = %v, want %v", result.Score, silverScore)
-			}
-		})
-		t.Run("gold", func(t *testing.T) {
-			r := &checker.CIIBestPracticesData{
-				Badge: clients.Gold,
-			}
-			result := CIIBestPractices("CIIBestPractices", nil, r)
-			if result.Score != checker.MaxResultScore {
-				t.Errorf("CIIBestPractices() = %v, want %v", result.Score, checker.MaxResultScore)
-			}
-		})
-		t.Run("not found", func(t *testing.T) {
-			r := &checker.CIIBestPracticesData{
-				Badge: clients.NotFound,
-			}
-			result := CIIBestPractices("CIIBestPractices", nil, r)
-			if result.Score != checker.MinResultScore {
-				t.Errorf("CIIBestPractices() = %v, want %v", result.Score, checker.MinResultScore)
-			}
-		})
-		t.Run("error", func(t *testing.T) {
-			r := &checker.CIIBestPracticesData{
-				Badge: clients.Unknown,
-			}
-			result := CIIBestPractices("CIIBestPractices", nil, r)
-			if result.Score != -1 {
-				t.Errorf("CIIBestPractices() = %v, want %v", result.Score, -1)
-			}
-		})
-		t.Run("nil response", func(t *testing.T) {
-			result := CIIBestPractices("CIIBestPractices", nil, nil)
-			if result.Score != -1 {
-				t.Errorf("CIIBestPractices() = %v, want %v", result.Score, -1)
+	t.Parallel()
+	tests := []struct {
+		name     string
+		findings []finding.Finding
+		result   scut.TestReturn
+	}{
+		{
+			name: "Badge not found",
+			findings: []finding.Finding{
+				{
+					Probe:   "hasOpenSSFBadge",
+					Outcome: finding.OutcomeNegative,
+					Values: map[string]int{
+						"Unsupported": 1,
```

Comment:
the `Unsupported` value is unused I think. Can probably get rid of it from this test? It's getting a 0 from the negative outcome. 

Should the Unsupported key be included as a separate test for a positive outcome finding?

Reply:
This is intended to be a test that ensures that Scorecard scores 0 for badge levels that it does not recognize.

Reply:
And right now the test isn't doing that, because it's quitting early once it sees the negative outcome

Reply:
Ah right. I would say that we make the outcome positive to test for unsupported probes. WDYT?

Reply:
Sure. But lets split this into 2.
1 to test negative outcome. 
1 to test unsupported level (which should probably be a `-1`, not a 0?)

Reply:
AFAICT, `"Unsupported"` corresponds to these lines:

https://github.com/ossf/scorecard/blob/68573209d6d7df7e44e67cd6ffb68abfff0e1b68/checks/evaluation/cii_best_practices.go#L43-L44

which means that a positive outcome with an unsupported badge should return 0 and not -1.

WDYT?

Reply:
... Since evaluation also has a check for `clients.Unknown`.

Reply:
the text probably needs changed?

Reply:
Changed in https://github.com/ossf/scorecard/pull/3520/commits/054e4a99898714df5417ddb611cc8f92c22de606